internal/controller/freight/origins_test.go

```diff
@@ -214,6 +220,54 @@ func TestGetDesiredOrigin(t *testing.T) {
 				return m, &m.ArgoCDAppUpdates[0]
 			},
 		},
+		{
+			name: "ArgoCDAppUpdate for multi-source app with different origins for various sources are correctly identified",
```

Comment:
I'm not sure this new case adds a lot of value. All the existing cases cover the two crucial behaviors we require when discovering the desired origin for a promotion mechanism -- that child mechanisms can inherit from or override their parents.